### ISSUE TITLE
flux oracle for the MSL modal V.I is not yet correctly constructed (factor-2 convention + reactive-residue fixture)

### ISSUE THREAD

**Issue by bk-squared** [bk-squared/rfx#525]
The modal V·I pair does not measure the guided power on a **bisecting** mesh — the flux identity that certifies the corrected voltage span holds only where `h_sub/dx` is integral. Measured today on merged main (`f95240f`).

## What was measured

The extractor-independent oracle for the modal voltage is

```
Re(V · conj(I)) / flux_spectrum  ==  1
```

Poynting flux is accumulated by a different kernel, so it shares no code with the V·I path. PR #516 measured it on the **aligned** mesh (dx = 84.67 µm, `h_sub/dx` = 3.0000) and got 1.006–1.009 with the corrected span. That mesh is the easy case: the rounding proxy and the rasterized trace node coincide there, so PR #516's F2 anchoring change is a *no-op* on it (measured bit-identical). The mesh class where F2 actually bit had never had this identity measured — the PR body says so verbatim ("expected ~1 by construction; not yet measured on this mesh").

Now measured. Single driven MSL port, far end CPML-absorbed, 5 planes × 6 frequencies, dx = 80 µm (`h_sub/dx` = 3.1750, trace rasterized at node 4 vs rounding proxy 3), `num_periods=30`:

| span | ratio range |
|---|---|
| **shipped** (trace-anchored, edges 0..3) | **0.5382 – 0.6935** |
| proxy / pre-F2 (edges 0..2) — control, same recorded fields | 0.3968 – 0.5117 |

Both spans, all planes, all frequencies. Monotone with frequency (0.69 at 2.0 GHz → 0.54 at 4.5 GHz) and mildly plane-dependent.

Preflight on this fixture, verbatim, quoted per repo rule:

```
[PREFLIGHT] MSL port 'msl_0': only 3 substrate cell(s) in z (h_sub=254µm, dx=80µm).
  Yee staircase at dielectric interface is O(dx) — Z0 staircase error >5% expected.
  Refine to dx ≤ 64µm (4+ substrate cells) for <5% Z0 bias.
[PREFLIGHT] MSL port 'msl_0': h_sub/dx = 3.175 (fractional part 0.175) lands in the
  [0.10, 0.40] mixed-cell danger zone. The substrate-air interface bisects the same
  Yee cell that holds the trace ... To snap onto a safe alignment, set dx = 63.5µm
  (= h_sub/4) or 84.7µm (= h_sub/3).
[PREFLIGHT] 'pec' z-extent 80µm = 1.0 cells — below 1 cell resolution ...
```

Script: `/tmp/verify_flux_bisecting.py` (to be committed under `scripts/diagnostics/` as part of #520's flux-oracle-artifact item).

## What this does and does not say about PR #516

**Does say**: the F2 anchoring fix is a real improvement even here — 0.54–0.69 against the proxy's 0.40–0.51, i.e. it recovers a large part of the deficit, consistent with it restoring a genuinely-missing substrate edge. And PR #516's aligned-mesh result stands unchanged (1.006–1.009, independently replicated by that PR's review with its own numpy re-derivation).

**Does not say**: that the extractor is correct on bisecting meshes. It is not, by this oracle. Roughly a third of the guided power is unaccounted for at 4.5 GHz. So the honest scope of the merged fix is **aligned meshes**; on bisecting meshes it is *better than before* and *still wrong*.

**Nothing here is a regression** — the pre-#516 code was worse on this mesh by the same measure.

## The gate fixture sits on the bad mesh

`tests/test_msl_thru_line_passive_gate` and its length-invariance sibling use dx = 80 µm, i.e. exactly this class. They pass, and their calibrations (0.1160 / 0.9930 / 57.58 Ω) are honest recordings of what the extractor produces there — but "passes the gate" and "the V·I pair measures the guided power" are now known to be different statements at that mesh. The gate docstring already warns that this fixture is on a mesh the repo's own preflight advises against; it should also carry this.

## Hypothesis (NOT verified — recorded so it is not mistaken for a finding)

On a bisecting mesh the ground→trace path crosses a Yee cell that is part dielectric, part air, and the trace floats 66 µm above the dielectric top. Summing that mixed cell's `Ez` as if it were a uniform edge of the modal voltage path is the obvious suspect, and the frequency trend is at least consistent with a geometry-fixed voltage error growing against a rising reference. But the only thing measured is that the ratio is not 1.

## Suggested first probes (cheap, one recorded run each)

1. **Third mesh class**: dx = 63.5 µm (`h_sub/dx` = 4.0, aligned). If the ratio returns to ~1 there, "integral `h_sub/dx`" is the controlling variable and the story is the mixed cell.
2. **Sweep the fraction**: a few dx values spanning `frac ∈ (0, 0.5)` — does the deficit scale with the fraction, i.e. with how much of the mixed cell is dielectric?
3. **Per-edge decomposition**: dump each Ez edge's contribution at one plane on the bisecting mesh and compare against a static-Laplace potential profile for the same rasterized geometry — that separates "the sum is over the wrong edges" from "the field in the mixed cell is not what a potential path integral needs".
4. **Current, not just voltage**: the oracle constrains the *product*. `msl_loop_current`'s contour on a bisecting mesh straddles the same mixed cell; it has not been isolated. A V-only hypothesis should not be assumed before the loop is checked independently.

Related: #518 (the invariance lock on this same mesh is margin-less), #519/#520 (superseded-extractor evidence chains), #487 (its floor advisory must key on the alignment class, not cell count).

**Comment by bk-squared** [comment]
## Probe 1 ran; its CONTROL failed, so its conclusion is withdrawn

Pre-declared before running: *"84.67 reproduces the known 1.006-1.009 as a control. If the control does not reproduce, this harness is wrong and nothing else here counts."*

It did not reproduce.

Span sweep, `Re(V·conj(I))/flux` per k_top, single driven port, num_periods=25, 3 planes:

```
dx = 84.67 um (ALIGNED, control)          dx = 80.00 um (BISECTING)
 k_top   2.0G     3.0G     4.0G            k_top   2.0G    3.0G    4.0G
     1  0.5068  -0.7773   1.4113               1  0.1613  0.1587  0.1426
     2  1.0225  -1.5614   2.8410               2  0.3263  0.3211  0.2888
     3  1.5472  -2.3514   4.2750  <-SHIPPED    3  0.4976  0.4896  0.4407
     4  1.3546  -3.5356   5.6380               4  0.6744  0.6636  0.5977  <-SHIPPED
     5  1.1770  -4.7316   7.0421               5  0.6589  0.6421  0.5610
```

The dx=80 column is smooth, monotone in k_top, and consistent across frequency — and it **reproduces my earlier independent measurement** on that mesh (0.54-0.69 there, ~0.60-0.67 here). So the harness is not uniformly broken.

The dx=84.67 column is **sign-flipping across frequency** (−0.78 … −4.73 at 3 GHz) and reaches 4.28. That is not a wrong voltage span — a span error scales V by a positive factor and cannot flip a sign. It is the signature of a contaminated **denominator**.

## Withdrawn

Probe 1's headline — *"no span reaches 1 on any mesh, therefore H-DEFN (the V·I definition is inadequate)"* — is **withdrawn for both meshes**. On a fixture whose flux reference is unsound, "no span reaches 1" is not evidence about spans.

Note the one thing probe 1 did establish, which is independent of the flux reference: the rasterized `eps_r` per z-node.

```
dx = 84.67 (h/dx = 3.0000):  3.66 3.66 3.66 | 1.00 1.00     trace node 3
dx = 80.00 (h/dx = 3.1750):  3.66 3.66 3.66 3.66 | 1.00     trace node 4
dx = 72.57 (h/dx = 3.5000):  3.66 3.66 3.66 3.66 | 1.00     trace node 4
dx = 63.50 (h/dx = 4.0000):  3.66 3.66 3.66 3.66 | 1.00     trace node 4
```

**This corrects the geometry story I posted earlier.** I wrote that on a bisecting mesh the trace "floats 66 µm above the dielectric with an air gap". Wrong: the substrate rasterizes to a **full extra cell** — 4 substrate nodes = 320 µm instead of the intended 254 µm — with the trace directly on top. A 26% thicker dielectric, no gap. That is a cleaner explanation of why dx=80 reads Z0 = 57.58 Ω against the aligned mesh's 44.11 Ω, and it makes "different rasterized structure" concrete rather than hand-wavy.

## Next probe (running)

Measure the denominator's own health before any further ratio is believed. On a lossless uniform line the net x-flux must be **plane-independent**; if it varies or flips sign, the record carries standing-wave/near-field contamination and no ratio against it means anything. Reporting flux per plane per frequency, its plane-to-plane spread, sign consistency, and a settling witness, at both meshes and at num_periods 25 and 60.

If the denominator is contaminated at dx=84.67 only, that also puts a question against the original 1.006-1.009 measurement — which was taken on a fixture I have not been able to reproduce end-to-end.

**Comment by bk-squared** [comment]
## Correction: this issue's numbers carry a factor-2 convention error, and the fixture cannot support the oracle anyway

Two defects in my own measurement, both now measured rather than argued. **The headline of this issue — "the modal V·I pair recovers only 0.54–0.69 of the guided power" — is withdrawn.**

### 1. The two instruments are the same; the 0.60 factor was mine

Direct comparison of the `FluxMonitor`'s own accumulators against DFT plane probes registered at the *same* coordinate (dx = 80 µm, x = 7.0 mm):

```
monitor index 96, axis 0, dA = [[6.3999996e-09]]     dx^2 = 6.4e-09      <- identical
e1_dft vs ey : sum|mon-probe|/sum|probe| = 0.000000   IDENTICAL
e2_dft vs ez : sum|mon-probe|/sum|probe| = 0.000000   IDENTICAL
h1_dft vs hy : sum|mon-probe|/sum|probe| = 0.004963   0.5% (half-step timing)
h2_dft vs hz : sum|mon-probe|/sum|probe| = 0.004932   0.5%
```

So my earlier speculation that the monitor samples different points or uses a different area weight is **refuted** — E is bit-identical, H differs by 0.5%, `dA` is exactly `dx²`.

And a float64 redo of `flux_spectrum`'s own formula from the monitor's internals reproduces the library **exactly** (ratio 1.00000 at both frequencies). The library is doing precisely what its source says.

### 2. The factor of 2 — the actual defect

`flux_spectrum` (`rfx/probes/probes.py:635-636`) returns

```python
integrand = mon.e1_dft * jnp.conj(mon.h2_dft) - mon.e2_dft * jnp.conj(mon.h1_dft)
flux = jnp.real(jnp.sum(integrand * mon.dA, axis=(-2, -1)))
```

— **no ½ prefactor.** It is `∫Re(E×H*)·dA`, which is *twice* the time-averaged power for phasor amplitudes.

Every ratio I posted in this issue divided **`0.5·Re(V·conj(I))`** — the time-*averaged* modal power — by that. Averaged over non-averaged. Measured directly:

```
2.0 GHz   library flux_spectrum          = 1.48956e-28
          hand Poynting, no ½ prefactor  = 1.77582e-28    ratio 1.192
          hand Poynting, with ½          = 8.87911e-29    ratio 0.596
```

So the correct comparison is `Re(V·conj(I)) / flux_spectrum`, and every number in this issue must be **doubled**:

| mesh | as posted | convention-corrected |
|---|---|---|
| dx = 80 µm (bisecting) | 0.54 – 0.69 | **1.20 – 1.34** |
| dx = 84.67 µm (aligned, PR #516's oracle) | 1.006 – 1.009 | **2.01 – 2.02**, *if* that measurement used the same convention |

The direction reverses: the modal pair does not under-report, it *over*-reports. And PR #516's "1.006–1.009 ⇒ the corrected V span is right" is only sound if that measurement paired ½ with ½ — which I cannot check, because its harness was never committed (it is the flux-oracle artifact item in #520, and this is exactly why that item matters).

### 3. The fixture cannot support this oracle regardless

Even with the convention fixed, this single-driven-port fixture is the wrong testbed. The trace runs into the x-CPML at the far end, and a CPML does not cleanly absorb a *guided* mode — so the plane carries a largely reactive field and the net flux is a small residue of much larger quantities. Evidence:

- at dx = 84.67 µm the net flux is ~1e-30 W with **472% / 147% / 304%** plane-to-plane spread and random sign, on a record settled to −122.9 dB — not under-settling, a standing wave;
- a **0.5%** difference between the two instruments' H accumulators (a half-step timing convention) moves the *real part* by **19%**, which only happens when `Re(E·H*)` is small against `|E·H*|`.

A quantity that swings 19% on a 0.5% input difference is not a usable reference.

### What a sound measurement needs

1. **Fix the convention**: compare `Re(V·conj(I))` against `flux_spectrum`, or `0.5·Re(V·conj(I))` against `0.5·flux_spectrum`. Not one of each.
2. **A fixture with a real termination** so the net flux is not a small reactive residue — a matched 2-port thru, or a line long enough with a lossy termination that the plane sees a travelling wave. Report `Re(E·H*)` versus `|E·H*|` at the plane as an admissibility witness before quoting any ratio.
3. **Commit the harness** (#520's flux-oracle artifact item). PR #516's number and mine disagree by exactly the factor this correction is about, and neither harness was committed, so the disagreement could not be diagnosed from the repo.

### Consequences for the merged fix

**PR #516's V-span correction is NOT invalidated by this** — it rests on more than the flux oracle: the Ez-edge derivation from the Yee curl staggering (independently re-derived in review), the trace-node rasterization measurement, the aligned/bisecting differential (2-cell stamp bit-unchanged, 1-cell moved 13.8%), and the committed gates. What *is* invalidated is my claim that the fix is "verified only on aligned meshes, and wrong on bisecting ones" — that claim came from this issue's numbers, and this issue's numbers were mis-normalised.

The honest state: **the flux oracle has not yet been correctly evaluated on any mesh.** Reopening as an oracle-construction task rather than an extractor defect. #527 (the AD-vs-FD gradient failure) is unaffected — it does not go through this oracle.
